examples/ui/flex_display.rs

```diff
@@ -0,0 +1,353 @@
+                                                            align_items: AlignItems::FlexStart,
+                                                            justify_content:
+                                                                JustifyContent::FlexStart,
+                                                            padding: UiRect {
+                                                                left: Val::Px(5.),
+                                                                top: Val::Px(5.),
+                                                                ..Default::default()
+                                                            },
+                                                            ..Default::default()
+                                                        },
+                                                        background_color: BackgroundColor(
+                                                            palette[3],
+                                                        ),
+                                                        ..Default::default()
+                                                    },
+                                                    ButtonTarget {
+                                                        id: target_ids.pop().unwrap(),
+                                                        color: palette[3],
+                                                    },
+                                                ))
+                                                .with_children(|parent| {
+                                                    parent.spawn((
+                                                        TextBundle {
+                                                            text: Text::from_section(
+                                                                "",
+                                                                text_style.clone(),
+                                                            ),
+                                                            ..Default::default()
+                                                        },
+                                                        BackgroundColor(Color::BLACK.with_a(0.5)),
+                                                    ));
+                                                });
+                                        });
+                                });
+                        });
+                });
+        });
+}
+
+fn update(
+    mouse: Res<Input<MouseButton>>,
+    keyboard: Res<Input<KeyCode>>,
+    mut left_query: Query<
+        (&mut BackgroundColor, &mut Style, &mut Visibility),
+        Without<Interaction>,
+    >,
+    mut right_query: Query<(&mut BackgroundColor, &ButtonTarget, &Interaction)>,
+) {
+    right_query.for_each_mut(|(mut background_color, button_target, interaction)| {
```

Comment:
we prefer `for (..) in &mut right_query` loops in example rather than using `for_each_mut`

Reply:
I'll change it.